### PR TITLE
VideoPlayer: move supported deinterlacing methods into ProcessInfo

### PR DESCRIPTION
**Pull request by FernetMenta**
fixes missing deinterlacing methods for sw decoding

@fritsch /cc

### CODE REVIEW

File: xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp [hunk 1]
```diff
@@ -180,6 +170,30 @@ float CProcessInfo::GetVideoDAR()
...
+  CSingleLock lock(m_videoCodecSection);
+
+  m_deintMethods = methods;
+}
+
+bool CProcessInfo::Supports(EINTERLACEMETHOD method)
+{
+  CSingleLock lock(m_videoCodecSection);
+
+  for (auto &deint : m_deintMethods)
```
Comment (fritsch) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp]: std::find ?

**Comment by fritsch** [comment]
Cool thx much. I will close mine.

jenkins build this please

File: xbmc/cores/VideoPlayer/Process/ProcessInfo.h [hunk 2]
```diff
@@ -57,6 +59,7 @@ class CProcessInfo
   int GetAudioSampleRate();
   void SetAudioBitsPerSample(int bitsPerSample);
   int GetAudioBitsPerSampe();
```
Comment (fritsch) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.h]: Ouh - if you just touch that file GetAudioBitsPerSample <- the L is missing.

File: xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp]: yes but internet is not always good here

File: xbmc/cores/VideoPlayer/Process/ProcessInfo.h (hunk 2 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.h]: :)

**Comment by fritsch** [comment]
Can you squash the first commit into yours? (No need to keep my authorship) - then we can get jenkins build and can finally merge?

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp [hunk 3]
```diff
@@ -180,6 +170,29 @@ float CProcessInfo::GetVideoDAR()
...
+  CSingleLock lock(m_videoCodecSection);
+
+  m_deintMethods = methods;
+}
+
+bool CProcessInfo::Supports(EINTERLACEMETHOD method)
+{
+  CSingleLock lock(m_videoCodecSection);
+
+  auto it = std::find(m_deintMethods.begin(), m_deintMethods.end(), method);
```
Comment (Jalle19) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp]: Could be just "return std::find() != end()"

Comment (fritsch) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp]: I had debugging in mind. But yes.

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/ProcessInfo.cpp]: never ever put multiple statements on a single line.

**Comment by fritsch** [comment]
jenkins is fine - let's fix that bug.

**Comment by fritsch** [comment]
That patches out too much. We now have removed the selection dialog for selecting _any_ deinterlacer :-)

**Pull request merged**